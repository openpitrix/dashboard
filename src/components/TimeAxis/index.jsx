import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import { formatTime } from 'utils';

import styles from './index.scss';

export default class TimeAxis extends PureComponent {
  static propTypes = {
    timeList: PropTypes.array
  };

  render() {
    const { timeList } = this.props;

    return (
      <div className={styles.timeAxis}>
        <ul className={styles.axis}>
          {timeList
            && timeList.map((item, index) => (
              <li key={index}>
                <div className={styles.word}>{item.job_action}</div>
                <div className={styles.time}>
                  {formatTime(item.create_time)}
                </div>
                <span className={classnames(styles.dot, styles[item.status])} />
              </li>
            ))}
        </ul>
      </div>
    );
  }
}
