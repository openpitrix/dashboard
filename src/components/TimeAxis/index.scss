@import '~scss/vars';

.timeAxis {
  margin-left: 10px;
  .axis{
    li{
      position: relative;
      padding-left: 28px;
      padding-bottom: 25px;
      border-left: 2px solid #e2d0f2;
      &:first-child .dot, &:last-child .dot{
        //background-color: #cc5252;
      }
    }
    .word{
      margin-bottom: 6px;
      line-height: 19px;
      font-size: 16px;
      font-weight: bold;
      color: #300e56;
    }
    .time{
      line-height: 16px;
      font-size: 14px;
      color: #333333;
    }
    .dot{
      display: inline-block;
      position: absolute;
      left: -7px;
      top: -1px;
      width: 12px;
      height: 12px;
      border-radius: 50%;
      background-color: #24b36b;
      border: 2px solid #ffffff;
    }
    .StopClusters,DeleteClusters{
      background-color: #cc5252;
    }
  }
}
