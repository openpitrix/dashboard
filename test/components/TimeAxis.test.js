import TimeAxis from 'components/TimeAxis';

describe('Component TimeAxis', () => {
  const timeList = [
    {
      job_action: 'CreateCluster',
      create_time: '2019-04-08T02:39:28Z',
      status: 'successful'
    },
    {
      job_action: 'UpgradeCluster',
      create_time: '2019-04-09T02:39:28Z',
      status: 'working'
    },
    {
      job_action: 'RollbackCluster',
      create_time: '2019-04-09T03:39:28Z',
      status: 'failed'
    }
  ];

  it('basic render', () => {
    const wrapper = render(<TimeAxis timeList={timeList} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
