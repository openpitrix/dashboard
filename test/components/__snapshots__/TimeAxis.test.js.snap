// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component TimeAxis basic render 1`] = `
<div
  class="timeAxis"
>
  <ul
    class="axis"
  >
    <li>
      <div
        class="title"
      >
        CreateCluster
      </div>
      <div
        class="time"
      >
        2019/04/08 10:39:28
      </div>
      <span
        class="dot successful"
      />
    </li>
    <li>
      <div
        class="title"
      >
        UpgradeCluster
      </div>
      <div
        class="time"
      >
        2019/04/09 10:39:28
      </div>
      <span
        class="dot working"
      />
    </li>
    <li>
      <div
        class="title"
      >
        RollbackCluster
      </div>
      <div
        class="time"
      >
        2019/04/09 11:39:28
      </div>
      <span
        class="dot failed"
      />
    </li>
  </ul>
</div>
`;
